File: src/my_project/easy_problems/from201to250/intersection_of_two_arrays_ii.py
from typing import List 

class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        
        common_list = nums1 + nums2 
        
        dict_nums1 = dict()
        dict_nums2 = dict()
        
        for i in range(len(common_list)):
            dict_nums1[common_list[i]] = 0
            dict_nums2[common_list[i]] = 0
            
        len_nums1: int = len(nums1)
        len_nums2: int = len(nums2)
            
        for i in range(len_nums1):
            dict_nums1[nums1[i]] += 1
            
        for i in range(len_nums2):
            dict_nums2[nums2[i]] += 1
            
            
        answer: list[int] = []
        
        if len_nums1 <= len_nums2:
            for key in dict_nums1.keys():

                answer += [key] * min(dict_nums1[key],dict_nums2[key])
        
        
        else:
            for key in dict_nums2.keys():
                
                answer += [key] * min(dict_nums1[key],dict_nums2[key])
            
        
        return answer
    
solution = Solution()

print(solution.intersect(nums1 = [1,2,2,1], nums2 = [2,2]))

print(solution.intersect(nums1 = [4,9,5], nums2 = [9,4,9,8,4]))
    
    

File: my_project_typescript/easy_problems_typescript/from1to50/intersectionOfTwoArraysII.ts
function intersect(nums1: number[], nums2: number[]): number[] {

    let commonList: Array<number> = nums1.concat(nums2)

    let objNums1: {[key:number]: number} = {}
    let objNums2: {[key:number]: number} = {}

    for (let i = 0; i < commonList.length; i++){
        objNums1[commonList[i]] = 0
        objNums2[commonList[i]] = 0
    }

    let lenNums1: number = nums1.length
    let lenNums2: number = nums2.length

    for (let i = 0; i < lenNums1; i++){
        objNums1[nums1[i]] += 1
    }

    for (let i = 0; i < lenNums2; i++){
        objNums2[nums2[i]] += 1
    }

    let answer: Array<number> = []
    
    if (lenNums1 <= lenNums2){
        for (let key in objNums1){
            
            answer = answer.concat(Array(Math.min(objNums1[key],objNums2[key])).fill(key))
        }
    }
    else{
        for (let key in objNums2){
            
            answer = answer.concat(Array(Math.min(objNums1[key],objNums2[key])).fill(key))
        }
    }



    return answer

};

console.log(intersect([1,2,2,1],  [2,2]))

console.log(intersect([4,9,5],  [9,4,9,8,4]))







